extensions/notebook/src/extension.ts

```diff
@@ -231,6 +251,87 @@ async function addCell(cellType: azdata.nb.CellType): Promise<void> {
 	}
 }
 
+async function updateBookMetadata(bookPath: string): Promise<void> {
+	try {
+		// get the toc file and update the contents
+		let tocFilePath: string = path.join(bookPath, '_data', 'toc.yml');
+		let result: string;
+		fsPromises.readFile(tocFilePath).then(data => {
+			result = data.toString();
+			let contentFolders: string[] = [];
+			let headers = result.match(/- header: [a-zA-Z0-9\\.\s-]+$/gm);
+			let urls = result.match(/- url: [a-zA-Z0-9\\.\s-]+$/gm);
+			let firstLevelUrls = result.match(/^(?:\s+$[\r\n]+)+(- url: [a-zA-Z0-9\\.\s-]+$[\r\n]+)/gm);
+			let title: string;
+			let replacedString: string;
+			if (firstLevelUrls || headers || urls) {
+				// there must only be one first level url
+				if (firstLevelUrls && firstLevelUrls.length === 1) {
```

Comment:
And if there's not exactly one? Should we be displaying an error in that case?

Reply:
No if there aren't we should go ahead and check for headers. It's likely that the book doesn't have one. If it does it becomes the first section.